examples/distributed/elementwise.py

```diff
@@ -0,0 +1,45 @@
+import numpy
+import cupy
+from cupy.cuda import Device, Stream
+from cupyx.distributed.array import distributed_array
+
+
+SIZE = 300 * 300
+
+SHAPE = (300, 300)
+
+INDEX_MAP_A = {
+    0: (slice(200)),  # arr[:200]
+    1: (slice(200, None)),  # arr[200:]
+}
+
+INDEX_MAP_B = {
+    0: (slice(None), slice(None, None, 3)),  # arr[:,  ::2]
+    1: (slice(None), slice(1, None, 2)),  # arr[:, 1::2]
+}
+
+data_a = numpy.arange(SIZE).reshape(SHAPE)
+data_b = data_a + 1
+
+a = distributed_array(data_a, INDEX_MAP_A)
+
+b = distributed_array(data_b, INDEX_MAP_B)
+
+with Device(0):
+    s0 = Stream()
+    s0.use()
+
+with Device(1):
+    s1 = Stream()
+    s1.use()
```

Comment:
Quality of life change: This sample assumes there exists 2 GPUs, we need a device count check at the top and do early exit.